core/decorators.py

```diff
@@ -1,34 +1,81 @@
+from collections.abc import Callable
 from functools import partial, wraps
+from typing import ParamSpecArgs, ParamSpecKwargs
 
+from django.http import HttpRequest
+from django.utils.text import slugify
 from django.views.decorators.cache import cache_page as dj_cache_page
 
 from core.models import Config
 
+VaryByFunc = Callable[[HttpRequest, ParamSpecArgs, ParamSpecKwargs], str]
+
+
+def vary_by_accept(request, *args, **kwargs) -> str:
+    """
+    Return a cache usable string token that is different based upon Accept
+    header.
+    """
+    accept = request.headers.get("accept", "text/html").lower()
+    if (
+        "application/json" in accept
+        or "application/ld" in accept
+        or "application/activity" in accept
+    ):
+        return "accept-ap"
+    elif "text/html" in accept:
+        return "accept-html"
+    # Fallback is potentially uncached
+    return slugify("".join(str(x) for x in request.accepted_types))
```

Comment:
@andrewgodwin I'm not sure how I feel about this fallback. We need to cache before a decision is made and the `Accept` header can be very arbitrary. Good enough or any suggestions on how to improve this?

Reply:
Thinking about this, we should just have a middleware that detects the accept header or not. Let me write one of those and it'll simplify things, as we only care about "AP or not", versus the true power of the Accept header.